src/http/proxy_wasm/ngx_http_proxy_wasm_dispatch.c

```diff
@@ -240,11 +243,23 @@ ngx_http_proxy_wasm_dispatch(ngx_proxy_wasm_exec_t *pwexec,
         }
     }
 
+#if (NGX_HAVE_UNIX_DOMAIN)
+    if (host->len < uds_prefix.len ||
+        !ngx_str_eq(p, uds_prefix.len, uds_prefix.data, uds_prefix.len))
+    {
+        p = ngx_strlchr(p, last, ':');
+
+        if (p) {
+            port = ngx_atoi(p + 1, last - p);
```

Comment:
There are better ways to use preprocessor directives that would avoid duplicating this port-extraction logic, all we need to do is not run this logic if we have `NGX_HAVE_UNIX_DOMAIN` and the `unix:` protocol in the host.